### ISSUE TITLE
Add more versions to protocol hack

### ISSUE THREAD

**Issue by LeafHacker** [ImpactDevelopment/ImpactClient#26]
#24 asked whether the 1.11.x Protocol Hack supports other Minecraft versions such as 1.8. This is a feature request to add as many versions as possible to the protocol hack.

This may require a new interface to replace the toggle button, depending on how many versions the hack works well for.

**Comment by TheForgottenKing** [comment]
This would probably work on 1.9/1.10 but the pvp would be broken unless there was a patch for that.

**Comment by LeafHacker** [comment]
Yes, obviously the more differences in version protocol the harder it would get to emulate older versions.

Realistically it may not even be worth it to emulate 1.11.2 from 1.12, depending on how different it is.

The PvP change in 1.9 is a big (and obvious) one, but there are likely hundreds of other things that would be version dependant.

**Comment by ZeroMemes** [comment]
Changes in the protocol across major versions (1.8 vs 1.9 for example) require modified packet decoding for support. Most minor versions (1.11 vs 1.11.2) decode packets the same, so these changed aren't required.

**Comment by CDAGaming** [comment]
Perhaps a Protocol Changer for 4.1 could be added for say switching between 1.12 and 1.12.1 as the .1 is a minor version as well

**Comment by LeafHacker** [comment]
Right now this isn't being looked into as the reward isn't really worth the effort required
